test/helpers.h

```diff
@@ -246,4 +242,21 @@ struct LibraryHelper
     static Library::Error loadxmldoc(Library &lib, const tinyxml2::XMLDocument& doc);
 };
 
+class SimpleTokenizer2 : public Tokenizer {
+public:
+    template<size_t size>
+    SimpleTokenizer2(const Settings &settings, ErrorLogger &errorlogger, const char (&code)[size], std::vector<std::string> &files)
+        : Tokenizer{settings, errorlogger}
+    {
+        PreprocessorHelper::preprocess(code, files, *this, errorlogger);
+    }
+
+    // TODO: get rid of this
+    SimpleTokenizer2(const Settings &settings, ErrorLogger &errorlogger, const char code[], std::vector<std::string> &files)
+        : Tokenizer{settings, errorlogger}
+    {
+        PreprocessorHelper::preprocess(code, files, *this, errorlogger);
+    }
+};
+
```

Comment:
This seems like a superficial change but in the quest of untangling the mess around the language handling in `TokenList` this will shoulder much more in upcoming changes. The application code is actually straight forward but the custom tokenizing in the tests complicates things a lot and keep giving me headaches.